src/averageAges.js

```diff
@@ -38,6 +47,18 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  let women = people.filter(obj => obj.sex === 'f');
+
+  if (arguments.length > 1) {
+    women = women.filter((obj) => {
+      return people.some((elem) => elem.mother === obj.name);
+    });
+  };
+
+  const age = women.map(obj => obj.died - obj.born);
+  const sum = age.reduce((a, b) => a + b, 0);
+
+  return sum / women.length;
```

Comment:
create separate helper function to calculate average age

you repeat it several time